lib/rubygems/defaults.rb

```diff
@@ -79,7 +79,7 @@ def self.default_specifications_dir
   #++
 
   def self.find_home
-    Dir.home.dup
+    File.realpath(Dir.home)
```

Comment:
What is the reason for the `realpath`? That is IMHO almost always wrong thing to do.

Reply:
We `realpath` all the other paths when comparing paths in `Gem::Uninstaller`. So in order for the comparisons to work as expected, the fix is to `realpath` all of them.

This is the whole point of the fix.

Reply:
Alternatively I can expand it in `Gem::Uninstaller` directly when comparing paths. Maybe that's less invasive.

Reply:
@deivid-rodriguez I think it makes sense to move the `File.realpath(...)` call into `Gem::Uninstaller`. It avoids having to assume the input has already been normalized.

Reply:
Done!